src/bam/record.rs

```diff
@@ -113,12 +113,19 @@ fn extranul_from_qname(qname: &[u8]) -> usize {
 impl Record {
     /// Create an empty BAM record.
     pub fn new() -> Self {
-        Record {
+        let mut record = Record {
             inner: unsafe { MaybeUninit::zeroed().assume_init() },
             own: true,
             cigar: None,
             header: None,
-        }
+        };
+        // Developer note: these are needed so the returned record is properly
+        // initialized as unmapped.
+        record.set_tid(-1);
+        record.set_pos(-1);
+        record.set_mpos(-1);
+        record.set_mtid(-1);
```

Comment:
I would argue that we should also set the record to be unmapped by default.  I mean, how can it be mapped if the tid/pos are -1?

Reply:
Sounds reasonable but I'd quickly compare third party implementations (htslib/htsjdk/noodles) and/or spec footnotes (lots of gotchas there).